aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -100,22 +98,31 @@ public static ModelProviderStep builder() {
      */
     Completable hydrate() {
         final List<Completable> hydrationTasks = new ArrayList<>();
-        List<ModelSchema> modelSchemas = new ArrayList<>(modelProvider.modelSchemas().values());
+        List<ModelSchema> modelSchemas = Observable.fromIterable(modelProvider.modelSchemas().values())
+            // Don't sync models that have an associated syncExpression of QueryPredicates.none().
+            .filter(schema -> !QueryPredicates.none().equals(queryPredicateProvider.getPredicate(schema.getName())))
+            .toList()
+            .blockingGet();
 
         // And sort them all, according to their model's topological order,
         // So that when we save them, the references will exist.
         TopologicalOrdering ordering =
             TopologicalOrdering.forRegisteredModels(modelSchemaRegistry, modelProvider);
         Collections.sort(modelSchemas, ordering::compare);
+
         for (ModelSchema schema : modelSchemas) {
-            hydrationTasks.add(createHydrationTask(schema));
+            QueryPredicate predicate = queryPredicateProvider.getPredicate(schema.getName());
+            if (!QueryPredicates.none().equals(predicate)) {
```

Comment:
Hmm, why does `MatchNoneQueryPredicate` get special treatment here? Should other things that don't meet the predicate logic filter still pass through? (i.e. `predicate.evaluate()`)

Reply:
(Same applies to `filter()` above)

Reply:
> why does MatchNoneQueryPredicate get special treatment here?

There is no way to serialize a `MatchNoneQueryPredicate`, and send it to AppSync.  This makes sense, because there's no point in making an API call if you know the the response will just be zero results.    Basically, we don't want to perform the sync query at all, to eliminate an unnecessary API call.

> (Same applies to filter() above)

Woops, I actually don't need to filter twice!   I had debated doing the filter here and above, but accidentally left both of them in the code.  I'm going to do the filter at the top, and remove this one.